### ISSUE TITLE
switch_case_alignment false positive when using explicit return on a switch expression

### ISSUE THREAD

**Issue by allenhumphreys** [realm/SwiftLint#5191]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

This is very similar to https://github.com/realm/SwiftLint/issues/5080, except it's a different case of the same problem. So I think this issue should be simple to convey.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.52.4
* Installation method used (Homebrew, CocoaPods, building from source, etc)? homebrew
* Which Xcode version are you using (check `xcodebuild -version`)? 15A5229h

```swift
// This triggers a violation:
```swift
... <other stuff that prevents using an implicit return>

return switch environment.state {
case .normal: inverseColor ? palette.genericWhite : palette.secondaryIndigo
case .pressed: inverseColor ? palette.neutral200 : palette.secondaryNavy
case .disabled: inverseColor ? palette.neutral300 : palette.secondarySlate
}
```
